openml/datasets/functions.py

```diff
@@ -259,17 +269,11 @@ def check_datasets_active(
     dict
         A dictionary with items {did: bool}
     """
-    dataset_list = list_datasets(status="all", data_id=dataset_ids)
-    active = {}
-
-    for did in dataset_ids:
-        dataset = dataset_list.get(did, None)
-        if dataset is None:
-            if raise_error_if_not_exist:
-                raise ValueError(f"Could not find dataset {did} in OpenML dataset list.")
-        else:
-            active[did] = dataset["status"] == "active"
-
+    datasets = list_datasets(status="all", data_id=dataset_ids, output_format="dataframe")
+    active = dict(datasets["status"] == "active")
```

Comment:
Would it maybe be better / more readable to do `(datasets["status"] == "active")["did"])`

Reply:
That isn't the same, what I do here is create a `pd.Series` which indicates for each dataset if it is active or not: `datasets["status"] == "active"`, which gives something like:
```
>>> d["status"]=='active'
2        True
3        True
4        True
5        True
6        True
```
 
I then call `dict(pd.Series)`, which turns the index into keys, and the values into values: `{..., 45573: True, 45574: True}`.


(in fact, the proposed method would lead to an error as the series can not be indexed by `"did"`.

Reply:
Thanks for the explanation. Sorry, I'm not good enough yet reading pandas :(